### PR TITLE
Vinai/json contains

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Adds the json_contains function

### CODE REVIEW

File: sql/expression/function/json_contains.go [hunk 1]
```diff
@@ -41,18 +44,131 @@ import (
...
+
+	return &JSONContains{args[0], args[1], args[2]}, nil
 }
 
 // FunctionName implements sql.FunctionExpression
 func (j JSONContains) FunctionName() string {
 	return "json_contains"
 }
+
+func (j JSONContains) Resolved() bool {
```
Comment (Hydrocharged) [inline sql/expression/function/json_contains.go:74]: All of these need to be pointer receivers. If you're checking that you're implementing an interface via a pointer (`var _ sql.FunctionExpression = (*JSONContains)(nil)` here) and returning a pointer from `New`, then all of your functions should be pointers as well.

File: sql/expression/function/json_contains.go [hunk 2]
```diff
@@ -41,18 +44,131 @@ import (
...
+
+func (j JSONContains) Children() []sql.Expression {
+	if j.Path != nil {
+		return []sql.Expression{j.JSONTarget, j.JSONCandidate, j.Path}
+	}
+
+	return []sql.Expression{j.JSONTarget, j.JSONCandidate}
+}
+
+func (j JSONContains) WithChildren(children ...sql.Expression) (sql.Expression, error) {
```
Comment (Hydrocharged) [inline sql/expression/function/json_contains.go:172]: Check if the count is the same. If you sent out 3 expressions in `Children`, then you should have 3 here too.

**Review by Hydrocharged — APPROVED** [review]
Overall looks good, but I've mentioned this pointer receiver issue before.

File: sql/expression/function/json_contains.go (hunk 1 above)
Comment (VinaiRachakonda) [reply sql/expression/function/json_contains.go:74]: Totally my bad. Will be more cautious next time

**Pull request merged**